clowder/utility/git_submodule_utilities.py

```diff
@@ -95,27 +89,25 @@ def _create_repo_herd_branch(self, url, remote, branch, default_ref, depth=0):
 
     def _submodules_clean(self):
         """Clean all submodules"""
+        self._submodule_command('foreach', '--recursive', 'git', 'clean', '-ffdx',
+                                error_msg=' - Failed to clean submodules')
+
+    def _submodule_command(self, *args, error_msg=' - submodule command failed'):
+        """Base submodule command"""
+
         try:
-            self.repo.git.submodule('foreach', '--recursive', 'git', 'clean', '-ffdx')
+            self.repo.git.submodule(*args)
```

Comment:
@jlane9 is `*args` a list? I think the GitPython command requires comma separated args that it basically passes through to the git command, so just passing a list probably won't work

Reply:
So this is actually a neat little python trick. With _*args_ and _**kwargs_ python stores these in a list, dict respectively. So if you want to accept any number arg or any keyword arg you place these. Now the same can be said of the reverse. You can take a list, dict and expand them out to args, kwargs respectively.

Ex:

```python

# Args/Kwargs
def foo(*args, **kwargs):
    if 'print' in kwargs:
        for arg in args:
            print(arg)

# Prints 
# 1
# 2
# 3
# 4
foo(1, 2, 3, 4, print=True)

# Expand Args/Kwargs
ints = [1,2,3,4]
settings = {'print': True}

# Prints same as above
foo(*ints, **settings)
```

Reply:
nevermind. after fixing that copy paste issue, the cocos2d test passed so this must be ok. nice work.

Reply:
ah, nice. that's cool 👍

Reply:
Here's official documentation on above. https://docs.python.org/3/tutorial/controlflow.html#unpacking-argument-lists